### PR TITLE
[go-gen] Add generation of metric server instantiation in main, and refactor main gen

### PR DESCRIPTION
**Pull request by willjvsmith**
* Adds generation of the code in the main functions to spin up a Prometheus metric server
  * For both auth and non-auth services
* Tidies up the main function generation
* Adds Ports to the `util.go` `config` struct

You will notice I've commented out metric imports ready for my next PR where I add the actual metric calls to the handlers.

### CODE REVIEW

**Comment by willjvsmith** [comment]
Something fishy is causing the integration tests to fail, I'm not certain, but I don't think it has anything to do with the contents of this PR.

The output of the compiler should be silenced, but for some reason isn't behaving that way.

**Review by Lewiky — APPROVED** [review]
Cool cool cool

File: src/main/scala/temple/generate/server/go/common/GoCommonMainGenerator.scala [hunk 1]
```diff
@@ -29,57 +29,102 @@ object GoCommonMainGenerator {
...
+    )
+
+  private def generateMetricsBlock(): String = {
+    val goroutineBody = mkCode.lines(
+      genMethodCall("http", "Handle", doubleQuote("/metrics"), genMethodCall("promhttp", "Handler")),
+      genMethodCall("http", "ListenAndServe", genMethodCall("fmt", "Sprintf", doubleQuote(":%d"), "promPort"), "nil"),
+    )
+    mkCode.lines(
+      "// Prometheus metrics",
+      genDeclareAndAssign(s"config.Ports[${doubleQuote("prometheus")}]", "promPort", "ok"),
```
Comment (jaylees14) [inline src/main/scala/temple/generate/server/go/common/GoCommonMainGenerator.scala:66]: `)}]` this is very pleasing

**Review by jaylees14 — APPROVED** [review]
Nice

**Pull request merged**